ui/servsettings.py

```diff
@@ -360,3 +386,322 @@ async def get_content(self):
+
+    @disnake.ui.button(label="Set Maximum", style=disnake.ButtonStyle.primary, row=1)
+    async def select_maximum(self, button: disnake.ui.Button, interaction: disnake.Interaction):
+        button.disabled = True
+        await self.refresh_content(interaction)
+        await interaction.send(
+            "Choose a new maximum roll total by sending a message in this channel. To reset it, respond with 'default'.",
+            ephemeral=True,
+        )
+        try:
+            input_msg: disnake.Message = await self.bot.wait_for(
+                "message",
+                timeout=60,
+                check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+            )
+            if input_msg.content.lower() == "default":
+                self.settings.randchar_max = None
+            else:
+                self.settings.randchar_max = int(input_msg.content)
+            with suppress(disnake.HTTPException):
+                await input_msg.delete()
+        except (ValueError, asyncio.TimeoutError):
+            await interaction.send("No valid maximum found. Press `Set Maximum` to try again.", ephemeral=True)
+        else:
+            await self.commit_settings()
+            await interaction.send("Your maximum score has been updated.", ephemeral=True)
+        finally:
+            button.disabled = False
+            await self.refresh_content(interaction)
+
+    @disnake.ui.button(label="Add Over/Under Rule", style=disnake.ButtonStyle.primary, row=1)
+    async def add_rule(self, button: disnake.ui.Button, interaction: disnake.Interaction):
+        button.disabled = True
+        await self.refresh_content(interaction)
+        await interaction.send(
+            "Choose a new required over by sending a message in this channel.\n"
+            "Please use the format 'number>score' or 'number<score', for example '1>15' for at least one over 15, or "
+            "'2<10' for at least two under 10.",
+            ephemeral=True,
+        )
+        try:
+            input_msg: disnake.Message = await self.bot.wait_for(
+                "message",
+                timeout=60,
+                check=lambda msg: msg.author == interaction.author and msg.channel.id == interaction.channel_id,
+            )
+            if ">" in input_msg.content:
+                rule_type = "gt"
+                amount, value = input_msg.content.split(">")
+                new_rule = RandcharRule(type=rule_type, amount=amount, value=value)
+            elif "<" in input_msg.content:
+                rule_type = "lt"
+                amount, value = input_msg.content.split("<")
+                new_rule = RandcharRule(type=rule_type, amount=amount, value=value)
+            else:
+                raise ValueError
+            if self.settings.randchar_rules is None:
+                self.settings.randchar_rules = []
+            self.settings.randchar_rules.append(new_rule)
```

Comment:
We should limit the maximum number of rules to 25, so we don't overflow the select below.